### PR TITLE
core: migrate stdcm to new infra

### LINKED ISSUE #4266: core: kt-infra: stdcm: finish migration
Last step of the migration: remove reference to legacy classes

### LINKED ISSUE #5041: core: kt-infra: migrate ScheduleMetadataExtractor

### LINKED ISSUE #5042: core: kt-infra: stdcm: migrate tests

### LINKED ISSUE #5043: core: kt-infra: migrate standalone simulation

### LINKED ISSUE #5061: tests: fuzzer needs to be updated
The requests generated by the fuzzer aren't quite right since we dropped the api service.

It also needs https://github.com/osrd-project/osrd/issues/5058 to be fixed to be reliable.

### PR DESCRIPTION
**Pull request by eckter**
Change summary:
* Migrate train simulation code to work with the new infra interfaces
* Migrate STDCM to work with the new infra interfaces
* Run the STDCM pathfinding on blocks instead of routes
* Migrate tests, using a new class to build simplified infras
* Cleanup some legacy code (not all of it)

Note: the STDCM changes are fairly minimal (mostly renames and simple replacements). Most of the changes are STDCM dependencies, especially standalone sim. This PR can be reviewed by someone who isn't familiar with the STDCM module. 

Parts of it have already been reviewed in https://github.com/osrd-project/osrd/pull/5017, but it sounded more realistic to squash everything before the rebase.

### CODE REVIEW

File: core/kt-osrd-utils/src/main/kotlin/fr/sncf/osrd/utils/DistanceRangeMapImpl.kt [hunk 1]
```diff
@@ -172,4 +199,4 @@ data class DistanceRangeMapImpl<T>(
             return res
         }
     }
-}
\ No newline at end of file
+}
```
Comment (Khoyo) [inline core/kt-osrd-utils/src/main/kotlin/fr/sncf/osrd/utils/DistanceRangeMapImpl.kt]: Me like :)

File: core/kt-osrd-sim-infra/build.gradle [hunk 2]
```diff
@@ -18,6 +18,7 @@ dependencies {
     // PLEASE ADD AND UPDATE DEPENDENCIES USING libs.versions.toml
     implementation project(':kt-fast-collections')
     implementation project(path: ':osrd-geom')
+    implementation project(path: ':envelope-sim')
```
Comment (multun) [inline core/kt-osrd-sim-infra/build.gradle]: I'd rather not have this dependency. It's only used in a single place for a java adapter that could be moved somewhere else

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/RawSignalingInfra.kt [hunk 3]
```diff
@@ -44,4 +44,14 @@ fun RawSignalingInfra.getLogicalSignalName(signal: LogicalSignalId): String? {
     return getPhysicalSignalName(getPhysicalSignal(signal))
 }
 
+/** Returns the length of a route  */
+@JvmName("getRouteLength")
+fun RawSignalingInfra.getRouteLength(route: RouteId): Distance {
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/RawSignalingInfra.kt]: I'd rather precompute this, just like we do for blocks

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/BlockInfraImpl.kt [hunk 4]
```diff
@@ -121,3 +122,21 @@ class BlockInfraImpl(
         return blockPool[block].length
     }
 }
+
+@JvmName("getBlockEntry")
+fun BlockInfra.getRouteEntry(rawInfra: RawInfra, block: BlockId): DirDetectorId {
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/BlockInfraImpl.kt]:
```suggestion
fun BlockInfra.getBlockEntry(rawInfra: RawInfra, block: BlockId): DirDetectorId {
```

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/BlockInfraImpl.kt [hunk 5]
```diff
@@ -121,3 +122,21 @@ class BlockInfraImpl(
...
+    val blockPath: StaticIdxList<ZonePath> = getBlockPath(block)
+    val firstZone: ZonePathId = blockPath[0]
+    return rawInfra.getZonePathEntry(firstZone)
+}
+
+@JvmName("getBlockExit")
+fun BlockInfra.getRouteExit(rawInfra: RawInfra, block: BlockId): DirDetectorId {
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/BlockInfraImpl.kt]:
```suggestion
fun BlockInfra.getBlockExit(rawInfra: RawInfra, block: BlockId): DirDetectorId {
```

**Review by axrolld — APPROVED** [review]
Physics part LGTM 🚀

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt [hunk 6]
```diff
@@ -227,10 +230,35 @@ class RawInfraImpl(
         return res
     }
 
+    override fun getTrackChunkElectricalProfile(
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt]: I don't think it should be part of this API as is:
 1) either electrical profiles are part of the sim-infra API, and there is no mapping argument
 2) or it's something entirely separate, which thus does not belong in the API implementation

I'd rather have option 2, as this implementation can easily be moved outside of sim-infra

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/utils/EnvelopeTrainPathUtils.kt [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+import fr.sncf.osrd.sim_infra.impl.NeutralSection
+import fr.sncf.osrd.utils.DistanceRangeMap
+import fr.sncf.osrd.utils.DistanceRangeMapImpl
+import fr.sncf.osrd.utils.units.Distance
+
+
+/**
+ * Builds the ElectrificationMap
+ */
+fun buildElectrificationMap(path: PathProperties): DistanceRangeMap<Electrification> {
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/utils/EnvelopeTrainPathUtils.kt]: this does not belong here. it creates a dependency from sim-infra to envelope-sim, which I would like to avoid at all costs. Please move it in another module, such as the main code base.

File: core/src/main/java/fr/sncf/osrd/api/ElectricalProfileSetManager.java [hunk 8]
```diff
@@ -1,7 +1,7 @@
 package fr.sncf.osrd.api;
 
 import com.squareup.moshi.JsonDataException;
-import fr.sncf.osrd.external_generated_inputs.ElectricalProfileMapping;
+import fr.sncf.osrd.kt_external_generated_inputs.ElectricalProfileMapping;
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/api/ElectricalProfileSetManager.java]: what's the point of renaming the module? is it not the same?

File: core/src/main/java/fr/sncf/osrd/api/pathfinding/RemainingDistanceEstimator.java [hunk 9]
```diff
@@ -77,15 +78,15 @@ public static double minDistanceBetweenSteps(
...
-        var res = Double.POSITIVE_INFINITY;
-        var point = blockOffsetToPoint(this.blockInfra, this.rawInfra, blockIdx, (long) offset);
+    public double apply(Integer blockIdx, long offset) {
+        var resMeters = Double.POSITIVE_INFINITY;
+        var point = blockOffsetToPoint(this.blockInfra, this.rawInfra, blockIdx, offset);
         for (var target : targets)
-            res = Double.min(res, point.distanceAsMeters(target));
-        return res + remainingDistance;
+            resMeters = Double.min(resMeters, point.distanceAsMeters(target));
+        return Distance.fromMeters(resMeters) + remainingDistance;
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/api/pathfinding/RemainingDistanceEstimator.java:90]: what's going on here? is the double return value used to store a distance in millimeters intentionally? That rings a bell but I'm not 100% sure

File: core/src/main/java/fr/sncf/osrd/api/pathfinding/constraints/ElectrificationConstraints.java [hunk 10]
```diff
@@ -51,7 +51,8 @@ private static Set<Pathfinding.Range> getBlockedRanges(RollingStock stock, PathP
                 var blockingRanges = neutralSections.complement().subRangeSet(voltageInterval).asRanges();
 
                 for (var blockingRange : blockingRanges)
-                    res.add(new Pathfinding.Range(blockingRange.lowerEndpoint(), blockingRange.upperEndpoint()));
+                    if (blockingRange.lowerEndpoint() < blockingRange.upperEndpoint())
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/api/pathfinding/constraints/ElectrificationConstraints.java]: I'm not sure I understand this one: how can that happen? I wish there were a comment explaining it

File: core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt [hunk 11]
```diff
@@ -4,8 +4,8 @@ import com.carrotsearch.hppc.IntArrayList
 import com.squareup.moshi.Json
 import fr.sncf.osrd.api.ConflictDetectionEndpoint.ConflictDetectionResult.Conflict
 import fr.sncf.osrd.api.ConflictDetectionEndpoint.ConflictDetectionResult.Conflict.ConflictType
-import fr.sncf.osrd.standalone_sim.result.ResultTrain.SpacingRequirement
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt:7]: there aren't any functional changes in this file, I'd rather not touch it if you don't mind

File: core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java [hunk 12]
```diff
@@ -1,70 +1,56 @@
...
+    public static Envelope computeMRSP(PathProperties path, RollingStock rollingStock, boolean addRollingStockLength,
+                                       String trainTag) {
         var builder = new MRSPEnvelopeBuilder();
-        var pathLength = 0.;
-        for (var r : ranges)
-            pathLength += r.getLength();
+        var pathLength = toMeters(path.getLength());
 
-        // add a limit for the maximum speed the hardware is rated for
+        // Add a limit for the maximum speed the hardware is capable of
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java]: that's not technically correct, the max speed in the rolling stock definition is not what the train is capable of, it's lower. It's the maximum operational speed.

File: core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java [hunk 13]
```diff
@@ -1,70 +1,56 @@
...
-                if (Double.isInfinite(speed) || speed == 0)
-                    continue;
+        var offset = addRollingStockLength ? rollingStock.length : 0.;
+        var speedLimits = path.getSpeedLimits(trainTag);
+        for (var speedLimit: speedLimits) {
+            // Compute where this limit is active from and to
+            var start = toMeters(speedLimit.getLower());
+            var end = Math.min(pathLength, offset + toMeters(speedLimit.getUpper()));
+            var speed = toMetersPerSecond(rangeMapEntryToSpeed(speedLimit));
+            if (speed != 0)
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java:47]: have you checked it does not change simulation results? not just here, in general

File: core/src/main/java/fr/sncf/osrd/standalone_sim/SignalProjection.kt [hunk 14]
```diff
@@ -2,54 +2,54 @@ package fr.sncf.osrd.standalone_sim
...
 
-    // Get the route path
-    // TODO: do it in the pathfinding
-    val routes = MutableStaticIdxArrayList<Route>()
-    for (javaRoute in trainPath.routePath) {
-        val route = rawInfra.routeMap[javaRoute.element.infraRoute]!!
-        routes.add(route)
-    }
+    // get a new generation route path
+    val routePath = toRouteIdList(PathfindingResultConverter.chunksToRoutes(rawInfra, KtToJavaConverter.toIntList(trainPath.chunks)))
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/standalone_sim/SignalProjection.kt]: I'm surprised we have to guess the route path, didn't we have it?

File: core/src/main/java/fr/sncf/osrd/stdcm/graph/AllowanceManager.java [hunk 15]
```diff
@@ -58,32 +60,32 @@ private double getNeededDelay(STDCMEdge oldEdge) {
...
+                    .setEnvelope(extractEnvelopeSection(totalEnvelope, toMeters(previousEnd), toMeters(end)))
                     .setForceMaxDelay(true)
                     .setWaypointIndex(edge.waypointIndex())
                     .findEdgeSameNextOccupancy(edge.timeNextOccupancy());
             if (prevEdge == null)
                 return null;
             previousEnd = end;
         }
-        assert Math.abs(previousEnd - totalEnvelope.getEndPos()) < 1e-5;
+        assert Math.abs(toMeters(previousEnd) - totalEnvelope.getEndPos()) < 1e-5;
```
Comment (multun) [inline core/src/main/java/fr/sncf/osrd/stdcm/graph/AllowanceManager.java]: are we sure the 1e-5 margin still holds?

File: core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraAdapter.kt [hunk 16]
```diff
@@ -271,36 +282,48 @@ private fun makeChunk(
...
+            Pair(range.neutralSectionAnnouncements, true),
+        )) {
+            for (entry in rangeMap.asMapOfRanges()) {
+                val legacyNeutralSection = entry.value
+                res.put(
+                    Distance.fromMeters(entry.key.lowerEndpoint()),
+                    Distance.fromMeters(entry.key.upperEndpoint()),
+                    SimNeutralSection(legacyNeutralSection.lowerPantograph, isAnnouncement)
+                )
+            }
```
Comment (multun) [inline core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraAdapter.kt:363]: this stuff seems unrelated

File: core/src/main/kotlin/fr/sncf/osrd/utils/JavaInteroperabilityTools.kt [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+import fr.sncf.osrd.sim_infra.api.Route
+import fr.sncf.osrd.sim_infra.api.RouteId
+import fr.sncf.osrd.utils.indexing.StaticIdxList
+import fr.sncf.osrd.utils.indexing.mutableStaticIdxArrayListOf
+import fr.sncf.osrd.utils.units.Speed
+
+/** Java can't handle generics with value classes, it's impossible to extract the underlying speed from a
+ * `RangeMapEntry` in pure java. This method handles this specific case.  */
+fun rangeMapEntryToSpeed(entry: DistanceRangeMap.RangeMapEntry<Speed>): Long {
+    return entry.value.millimetersPerSecond.toLong()
```
Comment (multun) [inline core/src/main/kotlin/fr/sncf/osrd/utils/JavaInteroperabilityTools.kt:12]: isn't entry.value a long in java?

**Review by multun — CHANGES_REQUESTED** [review]

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/RawSignalingInfra.kt (hunk 3 above)
Comment (Erashin) [reply core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/RawSignalingInfra.kt]: Commit to check, route length computed during `resolveReferences` pre `RawInfraBuilderImpl.build()` method.

File: core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt (hunk 11 above)
Comment (Erashin) [reply core/src/main/java/fr/sncf/osrd/conflicts/Conflicts.kt:7]: Ngl, i disagree. Otherwise we'll never clean up the imports. This should have been done in the PR modifying this file, absolutely, but it wasn't, and I'd rather correct it now than never.

File: core/src/main/kotlin/fr/sncf/osrd/utils/JavaInteroperabilityTools.kt (hunk 17 above)
Comment (Erashin) [reply core/src/main/kotlin/fr/sncf/osrd/utils/JavaInteroperabilityTools.kt:12]: In Java, if you have a `DistanceRangeMap<Speed>`, then the value of `DistanceRangeMap.RangeMapEntry<Speed> entry` is not converted to a long, it stays as Speed. Which is why it needs to be converted (and `toMetersPerSecond()` does not work, as Java expects a long as an argument, but would here get a Speed).

File: core/src/main/java/fr/sncf/osrd/api/ElectricalProfileSetManager.java (hunk 8 above)
Comment (Erashin) [reply core/src/main/java/fr/sncf/osrd/api/ElectricalProfileSetManager.java:4]: No idea, I don't see a lot of Kotlin package names starting with `kt_` either. @eckter any idea ?

File: core/src/main/java/fr/sncf/osrd/standalone_sim/SignalProjection.kt (hunk 14 above)
Comment (Erashin) [reply core/src/main/java/fr/sncf/osrd/standalone_sim/SignalProjection.kt]: I don't really understand the question. At this point, the only thing we have is the original `RJSTrainPath` (which contains the route path, but in terms of RJSRoute, which is not what we want here) and the `PathProperties`, which does not contain the route path, but contains the path's track chunks (from which we compute the route path here). Do you see another simpler way to do it ?

File: core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraAdapter.kt (hunk 16 above)
Comment (Erashin) [reply core/src/main/kotlin/fr/sncf/osrd/sim_infra_adapter/RawInfraAdapter.kt:363]: We're now taking into account both the actual neutral sections as well as the announced neutral sections. The change makes sense to me. What do you mean exactly ?

File: core/src/main/java/fr/sncf/osrd/standalone_sim/SignalProjection.kt (hunk 14 above)
Comment (multun) [reply core/src/main/java/fr/sncf/osrd/standalone_sim/SignalProjection.kt]: As discussed, we need to do something like `val routePath = rjsRoutePath.map { infra.getRouteByName(it.id) }`

File: core/src/main/java/fr/sncf/osrd/stdcm/graph/AllowanceManager.java (hunk 15 above)
Comment (Erashin) [reply core/src/main/java/fr/sncf/osrd/stdcm/graph/AllowanceManager.java]: Indeed, this should be replaced by 1e-2.

File: core/src/main/java/fr/sncf/osrd/api/pathfinding/constraints/ElectrificationConstraints.java (hunk 10 above)
Comment (eckter) [reply core/src/main/java/fr/sncf/osrd/api/pathfinding/constraints/ElectrificationConstraints.java]: 0-length ranges can happen because of the way we build the `RangeSet`, I believe this was already the case before but it wasn't caught. 

But I've fixed it by using closed intervals instead when building the set, it makes more sense. I'll change that `if` into an `assert`.

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt (hunk 6 above)
Comment (eckter) [reply core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt]: The purpose was to use the projection tools in PathProperties, but I agree that it's awkward here. I'll make the projection tools public and move this to the place it's called

File: core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java (hunk 13 above)
Comment (eckter) [reply core/src/main/java/fr/sncf/osrd/envelope_sim_infra/MRSP.java:47]: We didn't run a fuzzer + before/after comparaison, but these parts have lots of good unit tests. We also run the fuzzer looking for 5xx errors.

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt [hunk 18]
```diff
@@ -36,11 +40,27 @@ interface PathProperties {
...
+    val beginOffset: Distance
+}
+
+/** Wraps the method without returning an optional, which can't be handled in java for value types */
+@JvmName("getTrackLocationOffsetOrThrow")
+fun PathProperties.getTrackLocationOffsetOrThrow(location: TrackLocation): Distance {
+    val res = getTrackLocationOffset(location)
+    if (res != null)
+        return res
+    throw OSRDError(ErrorType.InvalidScheduleTrackLocationNotIncludedInPath)
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt]: The error type if specific to the call site of this function: not all callers may want to yield an InvalidScheduleTrackLocationNotIncludedInPath error. I'd rather move this helper in a context closer to its use, rather than have it in PathProperties.kt

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraBuilder.kt [hunk 19]
```diff
@@ -3,11 +3,61 @@ package fr.sncf.osrd.sim_infra.impl
 import fr.sncf.osrd.fast_collections.mutableIntArrayListOf
 import fr.sncf.osrd.geom.LineString
 import fr.sncf.osrd.reporting.exceptions.OSRDError
-import fr.sncf.osrd.sim_infra.api.*
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraBuilder.kt:6]: is there a reason why this was removed?

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraBuilder.kt [hunk 20]
```diff
@@ -3,11 +3,61 @@ package fr.sncf.osrd.sim_infra.impl
...
+import fr.sncf.osrd.sim_infra.api.Zone
+import fr.sncf.osrd.sim_infra.api.ZoneId
+import fr.sncf.osrd.sim_infra.api.ZonePath
+import fr.sncf.osrd.sim_infra.api.ZonePathId
+import fr.sncf.osrd.sim_infra.api.decreasing
+import fr.sncf.osrd.sim_infra.api.increasing
 import fr.sncf.osrd.utils.DirectionalMap
 import fr.sncf.osrd.utils.DistanceRangeMap
-import fr.sncf.osrd.utils.indexing.*
-import fr.sncf.osrd.utils.units.*
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraBuilder.kt:10]: same

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt [hunk 21]
```diff
@@ -3,10 +3,58 @@ package fr.sncf.osrd.sim_infra.impl
 import fr.sncf.osrd.geom.LineString
 import fr.sncf.osrd.reporting.exceptions.ErrorType
 import fr.sncf.osrd.reporting.exceptions.OSRDError
-import fr.sncf.osrd.sim_infra.api.*
-import fr.sncf.osrd.utils.*
-import fr.sncf.osrd.utils.indexing.*
-import fr.sncf.osrd.utils.units.*
```
Comment (multun) [inline core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraImpl.kt:9]: same

File: core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt (hunk 18 above)
Comment (eckter) [reply core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt]: The function as a whole is a java/kt artifact for one specific call site. It can't be located in a .java file (a nullable inline value type isn't something we can handle).

The alternative would be to throw a generic error which would then be instantly caught and rethrown as an OSRDError. I'm not quite convinced it's better but I don't have a strong opinion on this, what do you think?

Comment (multun) [reply core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt]: I found a way to make it work without a stub:

```diff
diff --git a/core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt b/core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt
index f348a94a3..bbf9b44e9 100644
--- a/core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt
+++ b/core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/api/PathProperties.kt
@@ -46,6 +46,7 @@ interface PathProperties {
     fun getLength(): Distance
     @JvmName("getTrackLocationAtOffset")
     fun getTrackLocationAtOffset(pathOffset: Distance): TrackLocation
+    @JvmName("getTrackLocationOffset")
     fun getTrackLocationOffset(location: TrackLocation): Distance?
     fun <T> getRangeMapFromUndirected(getData: (chunkId: TrackChunkId) -> DistanceRangeMap<T>): DistanceRangeMap<T>
 
@@ -54,15 +55,6 @@ interface PathProperties {
     val beginOffset: Distance
 }
 
-/** Wraps the method without returning an optional, which can't be handled in java for value types */
-@JvmName("getTrackLocationOffsetOrThrow")
-fun PathProperties.getTrackLocationOffsetOrThrow(location: TrackLocation): Distance {
-    val res = getTrackLocationOffset(location)
-    if (res != null)
-        return res
-    throw OSRDError(ErrorType.InvalidScheduleTrackLocationNotIncludedInPath)
-}
-
 /** Build a Path from chunks and offsets, filtering the chunks outside the offsets */
 @JvmName("buildPathPropertiesFrom")
 fun buildPathPropertiesFrom(
diff --git a/core/src/main/java/fr/sncf/osrd/railjson/parser/RJSStopsParser.java b/core/src/main/java/fr/sncf/osrd/railjson/parser/RJSStopsParser.java
index 08764692a..54f26e982 100644
--- a/core/src/main/java/fr/sncf/osrd/railjson/parser/RJSStopsParser.java
+++ b/core/src/main/java/fr/sncf/osrd/railjson/parser/RJSStopsParser.java
@@ -1,7 +1,5 @@
 package fr.sncf.osrd.railjson.parser;
 
-import static fr.sncf.osrd.sim_infra.api.PathPropertiesKt.getTrackLocationOffsetOrThrow;
-
 import fr.sncf.osrd.railjson.schema.schedule.RJSTrainStop;
 import fr.sncf.osrd.reporting.exceptions.ErrorType;
 import fr.sncf.osrd.reporting.exceptions.OSRDError;
@@ -28,9 +26,12 @@ public class RJSStopsParser {
             long positionMM;
             if (stop.position != null)
                 positionMM = Distance.fromMeters(stop.position);
-            else
-                positionMM = getTrackLocationOffsetOrThrow(path,
-                        RJSTrackLocationParser.parse(infra, stop.location));
+            else {
+                var offset = path.getTrackLocationOffset(RJSTrackLocationParser.parse(infra, stop.location));
+                if (offset == null)
+                    throw new OSRDError(ErrorType.InvalidScheduleTrackLocationNotIncludedInPath);
+                positionMM = offset.getMillimeters();
+            }
 
             if (positionMM > path.getLength())
                 throw new OSRDError(ErrorType.InvalidScheduleOutsideTrainStopPosition);
```

**Review by multun — APPROVED** [review]
